internal/context.go

```diff
@@ -0,0 +1,32 @@
+// Copyright © 2022 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package internal
+
+import (
+	"context"
+	"time"
+)
+
+// Borrowed from https://cs.opensource.google/go/x/tools/+/refs/tags/v0.1.10:internal/xcontext/xcontext.go
```

Comment:
Do we then need to include the copyright notice from xcontext.go?

Reply:
I'm honestly not sure. See, it's not copied 1:1, it's mostly the idea that is taken from there but adjusted slightly (parent context is not a named field, the function name is different). I added the link to the original to at least give the original authors _some_ credit. But honestly, if I were to tackle this problem myself I would have written it the same way (it's just a few lines, hard to do things in a wildly different way).

That said, better safe than sorry, I will add the disclaimer to the top of the file.